gc/gc-iceberg-files/src/main/java/org/projectnessie/gc/iceberg/files/IcebergFiles.java

```diff
@@ -140,9 +140,14 @@ public Stream<FileReference> listRecursively(StorageUri path) throws NessieFileI
       }
       return StreamSupport.stream(fileInfos.spliterator(), false)
           .map(
-              f ->
-                  FileReference.of(
-                      basePath.relativize(f.location()), basePath, f.createdAtMillis()));
+              f -> {
+                StorageUri location = StorageUri.of(f.location());
+                if (!location.isAbsolute()) {
+                  location = basePath.resolve("/").resolve(location);
```

Comment:
Isn't `basePath.relativize(location)` below equal to `location` in this case?
(Maybe I'm too tired ATM tho)